tests/e2e/specs/product-feed/product-feed-campaign-notice.test.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * External dependencies
+ */
+import { expect, test } from '@playwright/test';
+/**
+ * Internal dependencies
+ */
+import { clearOnboardedMerchant, setOnboardedMerchant } from '../../utils/api';
+import ProductFeedPage from '../../utils/pages/product-feed';
+import SetupAdsAccountsPage from '../../utils/pages/setup-ads/setup-ads-accounts.js';
+import { LOAD_STATE } from '../../utils/constants';
+
+test.use( { storageState: process.env.ADMINSTATE } );
+
+test.describe.configure( { mode: 'serial' } );
+
+/**
+ * @type {import('../../utils/pages/setup-ads/setup-ads-accounts').default} setupAdsAccounts
+ */
+let setupAdsAccounts = null;
+
+/**
+ * @type {import('../../utils/pages/product-feed').default} productFeedPage
+ */
+let productFeedPage = null;
+
+/**
+ * @type {import('@playwright/test').Page} page
+ */
+let page = null;
+
+test.describe( 'Product Feed Page', () => {
+	test.beforeAll( async ( { browser } ) => {
+		page = await browser.newPage();
+		productFeedPage = new ProductFeedPage( page );
+		setupAdsAccounts = new SetupAdsAccountsPage( page );
+		await Promise.all( [
+			setupAdsAccounts.mockAdsAccountsResponse( [] ),
+			productFeedPage.mockRequests(),
+			setOnboardedMerchant(),
+		] );
+	} );
+
+	test.afterAll( async () => {
+		await clearOnboardedMerchant();
+		await page.close();
+	} );
+
+	test.describe( 'No campaign', () => {
+		test.beforeAll( async () => {
+			await productFeedPage.fulfillAdsCampaignsRequest( [] );
+			await productFeedPage.goto();
+		} );
+
+		test( 'No active product and no campaign; Do not display campaign notice', async () => {
+			await expect(
+				page.getByRole( 'heading', { level: 1, name: 'Product Feed' } )
+			).toBeVisible();
+
+			await productFeedPage.fulfillProductStatisticsRequest( {
+				timestamp: 1695011644,
+				statistics: {
+					active: 0,
+					expiring: 0,
+					pending: 0,
+					disapproved: 0,
+					not_synced: 1137,
+				},
+				scheduled_sync: 0,
+				loading: false,
+			} );
+
+			await page.reload();
+			await page.waitForLoadState( LOAD_STATE.DOM_CONTENT_LOADED );
```

Comment:
It doesn't seem to need reloading.

Reply:
We need it to update the statistics.

Reply:
Reload was required to update the stats.

Reply:
This gave a boost. The test ran much faster after this change. Thanks, @eason9487, for this suggestion.

Reply:
Done.